src/jit/codegenxarch.cpp

```diff
@@ -8842,7 +8966,80 @@ void CodeGen::genPutArgStk(GenTreePtr treeNode)
 #endif // !_TARGET_X86_
 }
 
-#if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+#if defined(FEATURE_PUT_STRUCT_ARG_STK)
+// genStoreRegToStackArg: Store a register value into the stack argument area
+//
+// Arguments:
+//    type   - the type of value to be stored
+//    reg    - the register containing the value
+//    offset - the offset from the base (see Assumptions below)
+//
+// Notes:
+//    A type of TYP_STRUCT instructs this method to store a 16-byte chunk
+//    at the given offset (i.e. not the full struct).
+//
+// Assumptions:
+//    The caller must set the context appropriately before calling this method:
+//    - On x64, m_stkArgVarNum must be set according to whether this is a regular or tail call.
+//    - On x86, the caller must set m_pushStkArg if this method should push the argument.
+//      Otherwise, the argument is stored at the given offset from sp.
+//
+// TODO: In the below code the load and store instructions are for 16 bytes, but the
+//          type is EA_8BYTE. The movdqa/u are 16 byte instructions, so it works, but
+//          this probably needs to be changed.
+//
+void CodeGen::genStoreRegToStackArg(var_types type, regNumber srcReg, unsigned offset)
+{
+    assert(srcReg != REG_NA);
+    instruction ins;
+    emitAttr    attr;
+    unsigned    size;
+    if (type == TYP_STRUCT)
+    {
+        ins = INS_movdqu;
+        // This should be changed!
+        attr = EA_8BYTE;
+        size = 16;
+    }
+    else
+    {
+#ifdef FEATURE_SIMD
+        if (varTypeIsSIMD(type))
+        {
+            assert(genIsValidFloatReg(srcReg));
+            ins = ins_Store(type);
+        }
+        else
+#endif // FEATURE_SIMD
+#ifdef _TARGET_X86_
+             if (type == TYP_LONG)
```

Comment:
I *thought* that the formatter liked to indent the if after an else in this situation (where the else is conditionally compiled). I tried eliminating the spaces, and jit-format puts them back. (Of course next time I push there will be a few more formatting changes!)

---
In reply to: [79509374](https://github.com/dotnet/coreclr/pull/7252#discussion_r79509374) [](ancestors = 79509374)